src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/toolbar_js.html.twig

```diff
@@ -455,60 +457,32 @@
+                        sfwdt.classList.add('sf-toolbar-opened');
+                        sfwdt.classList.remove('sf-toolbar-closed');
                     }
                 },
 
                 hideToolbar: function(token) {
                     var sfwdt = this.getSfwdt(token);
-                    addClass(sfwdt, 'sf-display-none');
+                    sfwdt.classList.add('sf-toolbar-closed');
+                    sfwdt.classList.remove('sf-toolbar-opened');
                 },
 
                 initToolbar: function(token) {
                     this.showToolbar(token);
 
-                    var hideButton = document.getElementById('sfToolbarHideButton-' + token);
-                    var hideButtonSvg = hideButton.querySelector('svg');
-                    hideButtonSvg.setAttribute('aria-hidden', 'true');
-                    hideButtonSvg.setAttribute('focusable', 'false');
-                    addEventListener(hideButton, 'click', function (event) {
+                    var toggleButton = document.querySelector(`#sfToolbarToggleButton-${token}`);
+                    addEventListener(toggleButton, 'click', function (event) {
                         event.preventDefault();
 
-                        var p = this.parentNode;
-                        p.style.display = 'none';
-                        (p.previousElementSibling || p.previousSibling).style.display = 'none';
-                        document.getElementById('sfMiniToolbar-' + token).style.display = 'block';
-                        setPreference('toolbar/displayState', 'none');
-                    });
-
-                    var showButton = document.getElementById('sfToolbarMiniToggler-' + token);
-                    var showButtonSvg = showButton.querySelector('svg');
-                    showButtonSvg.setAttribute('aria-hidden', 'true');
-                    showButtonSvg.setAttribute('focusable', 'false');
-                    addEventListener(showButton, 'click', function (event) {
-                        event.preventDefault();
-
-                        var elem = this.parentNode;
-                        if (elem.style.display == 'none') {
-                            document.getElementById('sfToolbarMainContent-' + token).style.display = 'none';
-                            document.getElementById('sfToolbarClearer-' + token).style.display = 'none';
-                            elem.style.display = 'block';
-                        } else {
-                            document.getElementById('sfToolbarMainContent-' + token).style.display = 'block';
-                            document.getElementById('sfToolbarClearer-' + token).style.display = 'block';
-                            elem.style.display = 'none'
-                        }
-
-                        setPreference('toolbar/displayState', 'block');
+                        const newState = 'opened' === getPreference('toolbar/displayState') ? 'closed' : 'opened';
```

Comment:
shouldn't this check the current state of the toolbar with `hasClass` instead ? This might do weird things for someone having the *old* preference (being `block` or `none`) in their localStorage